jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -210,25 +221,34 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
         # defaults to ``No Subset``). Since the Cubeviz parser puts the fluxes
         # and uncertainties in different glue Data objects, we translate the spectral
         # cube and its uncertainties into separate NDDataArrays, then combine them:
-        if self.aperture.selected != self.aperture.default_text:
+        if self.aperture.selected != self.aperture.default_text and self.wavelength_dependent:
             nddata = spectral_cube.get_subset_object(
                 subset_id=self.aperture.selected, cls=NDDataArray
             )
             uncertainties = uncert_cube.get_subset_object(
                 subset_id=self.aperture.selected, cls=StdDevUncertainty
             )
+            mask = self.cone_aperture()
+
+        elif self.aperture.selected != self.aperture.default_text:
+            nddata = spectral_cube.get_subset_object(
+                subset_id=self.aperture.selected, cls=NDDataArray
+            )
+            uncertainties = uncert_cube.get_subset_object(
+                subset_id=self.aperture.selected, cls=StdDevUncertainty
```

Comment:
I noticed during testing that this (and the same in the code blocks above and below) will fail if there is no uncertainty cube loaded. I think we want to allow that case and have the uncertainties be `None`